scripts/swupd-performance/swupdservicesetup.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/bash
+
+set -e
+
+echo "Setup staging directory"
+
+if [ -d "$HOME/staging" ]; then
+	echo "Staging directory already exists" || true
+else
+	dir=$HOME/staging
+	mkdir $HOME/staging
+
+	if [ $? -eq 0 ]; then
+		echo "Staging directory: $dir successfully created!"
+	fi
+fi
+
+echo ""
+echo "Setup performance monitoring script config dir"
+
+if [ -d /etc/swupd_perform_update_config ]; then
+	echo "Config directory already exists" || true
+else
+	mkdir /etc/swupd_perform_update_config
+fi
+
+cp -r stagingfilegen.sh /etc/swupd_perform_update_config/
+
+# Setting the systemd for the swupd-update-test.service
+cat <<EOF > /etc/systemd/system/swupd-update-test.service
+[Unit]
+Description=Update Software content
+
+[Service]
+Type=oneshot
+ExecStartPre=/etc/swupd_perform_update_config/stagingfilegen.sh
+EnvironmentFile=/etc/swupd_perform_update_config/script_config
+ExecStart=/usr/bin/swupd update --json-output --time
+StandardOutput=\$OUTFILE
+EOF
+
+# Setting timer for swupd-update-timer.service
+cat <<EOF > /etc/systemd/system/swupd-update-test.timer
+Description=Automatically Install Software Updates
+Documentation=man:swupd_update
+
+[Timer]
+OnBootSec=45min
+OnUnitActiveSec=45min
+AccuracySec=300
+RandomizedDelaySec=2700
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+echo "Setup of performance monitoring script config dir was successful!"
+
+echo ""
+echo "Perform system daemon reload"
+
+sudo systemctl daemon-reload
+
+if [ $? -eq 0 ]; then
+	echo "systemctl daemon-reload was successful!"
+fi
+
+echo ""
+echo "Perform swupd-update restart"
+sudo systemctl restart swupd-update-test.service
+
+if [ $? -eq 0 ]; then
+	echo "swupd-update-test.service was successfully scheduled!"
+fi
```

Comment:
you are using "set -e" at the beginning of the script but are never returning the shell to the initial state

Reply:
you mean like a rolllback to clean everything else?

Reply:
I was referring to using `set +e` at the end of the script to return the shell to the original state, but as a second thought since this will run in a sub shell it may not be necessary, but it won't hurt to add it either